halyard-cli/src/main/java/com/netflix/spinnaker/halyard/cli/command/v1/config/deploy/sizing/ComponentSizingEditCommand.java

```diff
@@ -0,0 +1,139 @@
+/*
+ * Copyright 2018 Google, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License")
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ *
+ */
+
+package com.netflix.spinnaker.halyard.cli.command.v1.config.deploy.sizing;
+
+import com.beust.jcommander.Parameter;
+import com.beust.jcommander.Parameters;
+import com.netflix.spinnaker.halyard.cli.command.v1.NestableCommand;
+import com.netflix.spinnaker.halyard.cli.command.v1.config.AbstractConfigCommand;
+import com.netflix.spinnaker.halyard.cli.services.v1.Daemon;
+import com.netflix.spinnaker.halyard.cli.services.v1.OperationHandler;
+import com.netflix.spinnaker.halyard.cli.ui.v1.AnsiUi;
+import com.netflix.spinnaker.halyard.config.model.v1.node.CustomSizing;
+import com.netflix.spinnaker.halyard.config.model.v1.node.DeploymentEnvironment;
+import lombok.AccessLevel;
+import lombok.Getter;
+
+import java.util.HashMap;
+import java.util.Map;
+
+@Parameters(separators = "=")
+public class ComponentSizingEditCommand extends AbstractConfigCommand {
+    @Getter(AccessLevel.PROTECTED)
+    private Map<String, NestableCommand> subcommands = new HashMap<>();
+
+    @Getter(AccessLevel.PUBLIC)
+    private String commandName = "edit";
+
+    @Getter(AccessLevel.PUBLIC)
+    private String serviceName;
+
+    @Parameter(
+            names = "--replicas",
+            description = "Set the number of replicas (pods) to be created for this service."
+    )
+    private Integer replicas;
```

Comment:
does this default to 1?

Reply:
In Java the default value for `Integer` is 0

Reply:
Oh I meant if it's not set will halyard still deploy 1 pod?

Reply:
I think it might do 0 pods, because it will be set to the default value of 0 and then `putIfNotNull` will be ran and the value is not null, so 0 will be placed for replicas. But I will test this out to see

Reply:
I think we're good, if this field isn't provided in the normal deployment, it'll default to 1.

Defaultly, it doesn't write it to `~/.hal/config`

Reply:
The Java default for `int` is 0, but the default for `Integer` is null [1]. Given we're using `Integer`, we should be fine because of the `putIfNotNull` call. For more clarity though, I could just put a default `1` here. What do you think?


https://stackoverflow.com/questions/2437603/why-does-using-a-default-valued-java-integer-result-in-a-nullpointerexception

Reply:
@costimuraru I think it's better to be explicit then implicit on this. I would default it to 1

Reply:
Made the change to default it to 1, @bpowell